api/src/main/java/org/openmrs/ui/framework/resource/ResourceFactory.java

```diff
@@ -62,6 +60,11 @@ public static ResourceFactory getInstance() {
 	 * @return the requested resource, from the requested provider, or null if not found 
 	 */
 	public File getResource(String providerName, String resourcePath) {
+		if (resourcePath == null || !resourcePath.equals(FilenameUtils.normalize(resourcePath)) || new File(resourcePath)
+				.isAbsolute()) {
+			return null;
+		}
+
```

Comment:
I understand this fix here and am supportive of it, but can you confirm that it does not prevent behaviors that are currently relied upon from working?  The main thing I am thinking of is "development mode" which allows for hot reloading of pages within UI framework when "watching" certain modules.  Does this affect that?  @mogoodrich  do you know?

Reply:
No sure @mseaton  @ibacher  we'd have to do some testing (and make sure this doesn't break anything else)

Reply:
So I'm reasonably sure this doesn't break live reloading in development mode. Development mode seems to be support not by providing absolute paths at the point where the resource is requested so much as by scanning a fixed set of development directories in the ModuleResourceProvider class. As long as the paths provided are relative paths (which they should be) that don't involve any directory traversal, things should be fine. So things like: `htmlforms:scripts/mygreatscript.js` will still load.

I'm hoping that the need to request a resource by an absolute path is... non-existent, but we'll have to see.

Reply:
Yeah, I think we can go with the assumption that the need to request by absolute path is non-existent and not allowed...

Reply:
@ibacher are you able to quickly test live reloading?

Reply:
@mogoodrich Testing this really quickly and it works fine (edited the htmlForm.js file).

Reply:
Great, thanks!